cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -368,11 +369,18 @@ public void performAction(HttpRequest request, HttpResponder responder,
     }
 
     if (!isValidAction(action)) {
-      responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
```

Comment:
I was following the existing code. The error handler is - as far as I understand - a fallback for exception thrown by methods further in the call stack. That does not prevent us from explicitly sending a response, does it?